test/e2e/e2e.go

```diff
@@ -43,6 +45,19 @@ func RunE2ETests(t *testing.T) {
 }
 
 var _ = Describe("validation", func() {
+	Context("Platform Check", func() {
+		It("should be either Kubernetes or OpenShift platform", func() {
+			cfg := ctrl.GetConfigOrDie()
+			platforminfo, err := platform.GetPlatformInfo(cfg)
+			Expect(err).ToNot(HaveOccurred())
+			if platforminfo.IsOpenShift() {
+				Expect(platform.GetPlatformName(cfg)).Should(Equal("OpenShift"))
+			} else {
+				Expect(platform.GetPlatformName(cfg)).Should(Equal("Kubernetes"))
```

Comment:
I'm not sure this part serves any purpose (other than checking that the consts declared here https://github.com/metallb/metallb-operator/blob/786de6242e351f3ca0515a0c180727f7802168ea/pkg/platform/types.go match "Openshift/Kubernetes"). Am I missing something?

Reply:
If we want to validate platform.GetPlatformInfo , I'd rather have the expected platform to be set with an environment variable. Something that defaults to Kubernetes but that we can change if we consume this tests in the openshift version too.

Something like this https://github.com/openshift-kni/cnf-features-deploy/blob/master/cnf-tests/testsuites/pkg/sriov/sriov.go#L17 in init() with the default being Kubernetes would make the trick.

Reply:
I wanted to add some basic test in e2e for platform pkg so I read the platform info and match with expected two strings if u have better suggestion pls let me know

Reply:
What I am suggesting in https://github.com/metallb/metallb-operator/pull/44#discussion_r668691257 is to check that the platform is what we expect from outside.
A boolean env variable that is false by default, that we raise when running the tests against OCP, so we know in advance what to expect by the result of IsOpenshift() and we validate that that check works.

Reply:
ok